### PR TITLE
add `deps` to `useSelector` to allow controlling stability of selector

### PR DESCRIPTION
**Pull request by MrWolfZ**
This PR adds a `deps` parameter to `useSelector` as discussed [here](https://github.com/reduxjs/react-redux/pull/1248#issuecomment-485335594).

It also fixes an issue with staleness of the selector.

### CODE REVIEW

File: test/hooks/useSelector.spec.js [hunk 1]
```diff
@@ -165,15 +165,40 @@ describe('React', () => {
...
+          // does not cause a re-render
+          store.dispatch({ type: '' })
+
+          expect(renderedItems).toEqual([0, 0])
+        })
       })
 
       describe('edge cases', () => {
         it('ignores transient errors in selector (e.g. due to stale props)', () => {
-          // TODO Not sure this test is really testing what we want.
```
Comment (MrWolfZ) [inline test/hooks/useSelector.spec.js]: This test was correct in what it tested. In fact, it failing showed an error in the implementation.

File: src/hooks/useSelector.js [hunk 2]
```diff
@@ -46,13 +52,15 @@ export function useSelector(selector) {
...
+
   const latestSubscriptionCallbackError = useRef()
-  const latestSelector = useRef(selector)
+  const latestSelector = useRef(memoizedSelector)
 
   let selectedState = undefined
 
   try {
-    selectedState = latestSelector.current(store.getState())
+    selectedState = memoizedSelector(store.getState())
```
Comment (MrWolfZ) [inline src/hooks/useSelector.js]: using `latestSelector.current` was wrong, since it was an stale selector. This change fixes that issue, which makes the test for stale props succeed.

**Comment by Dudeonyx** [comment]
If no dependency array is passed what happens?

File: src/hooks/useSelector.js (hunk 2 above)
Comment (markerikson) [reply src/hooks/useSelector.js]: Hah, nice catch :)

**Comment by markerikson** [comment]
@Dudeonyx : the latest passed-in selector should get used, instead of a previous one.

**Comment by markerikson** [comment]
LGTM.  I'll try to get alpha.1 published shortly.
